### PR TITLE
Fix appliance-repart-image NixOS VM test

### PR DESCRIPTION
**Pull request by jmbaur**
## Description of changes

With the update of systemd to v255, the repart tool switched to use 4K sector sizes by default. OVMF seems to not support sector sizes that aren't 512B, so we set the repart test to use 512B instead.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by malt3 — APPROVED** [review]

**Comment by jmbaur** [comment]
@ofborg test appliance-repart-image

**Comment by arianvp** [comment]
This sounds like the wrong fix. The test is showing a real failure! Namely that we can't boot images that repart poops out. Ovmf is widely used. If ovmf can't boot these images then that's a true failure. We shouldn't change the test to hide this. We should fix systemd or ovmf instead.

**Comment by nikstur** [comment]
I generally agree with @arianvp.

 @jmbaur are you sure that ovmf is the culprit? Maybe we just need to add an option to the qemu blockdevice declaration? I'll look into this soon too.

**Comment by jmbaur** [comment]
I agree as well, not 100% sure ovmf is the issue, though having the sector size option is helpful outside of this issue. I briefly tried using physical_block_size and logical_block_size at 4096 for the qemu disk device, which did not work. Haven't had time to debug further yet.

**Comment by nikstur** [comment]
However much I don't like it, I think reducing the sector size to 512 is the only sensible thing we can do. It seems like this a vfat issue. See [this comment](https://github.com/systemd/mkosi/blob/main/mkosi/__init__.py#L2772) in mkosi for background.

I think in this case we should set 512 as the default sector size for the sectorSize option. 4k sector size seems like an optimization that you should only enable if you need it explicitly.

File: nixos/modules/image/repart.nix [hunk 1]
```diff
@@ -135,6 +135,16 @@ in
       '';
     };
 
+    sectorSize = lib.mkOption {
+      type = with lib.types; nullOr (enum [ 512 1024 2048 4096 ]);
+      default = null;
```
Comment (nikstur) [inline nixos/modules/image/repart.nix:140]:
```suggestion
      default = 512;
```

File: nixos/modules/image/repart.nix [hunk 2]
```diff
@@ -135,6 +135,16 @@ in
       '';
     };
 
+    sectorSize = lib.mkOption {
+      type = with lib.types; nullOr (enum [ 512 1024 2048 4096 ]);
```
Comment (nikstur) [inline nixos/modules/image/repart.nix:139]:
```suggestion
      type = with lib.types; nullOr lib.types.int;
```

**Review by nikstur — CHANGES_REQUESTED** [review]

**Comment by jmbaur** [comment]
I've verified that repart is creating a valid 4K sector size image, the image can be used to create a loopback device with `losetup --sector-size=4096` and used just fine.

**Review by nikstur — APPROVED** [review]

**Comment by nikstur** [comment]
@ofborg test appliance-repart-image

**Review by WilliButz — APPROVED** [review]

**Pull request merged**